samples/check_tga/src/winmain.cpp

```diff
@@ -0,0 +1,90 @@
+#include <Windows.h>
+#include <windowsx.h>
+
+#include <rdmn-engine/logger/trivial_logger.hpp>
+
+#include <rcbe-engine/core/AbstractInputManager.hpp>
+#include <rcbe-engine/core/EditorInputManager.hpp>
+#include <rcbe-engine/core/WindowManager.hpp>
+#include <rcbe-engine/datamodel/system/window_config.hpp>
+#include <rcbe-engine/datamodel/visual/Texture.hpp>
+#include <rcbe-engine/parsers/tga/tga_parser.hpp>
+#include <rcbe-engine/utils/output_utils.hpp>
+
+#include <thread>
+
+int APIENTRY wWinMain(_In_ HINSTANCE hInstance,
+                     _In_opt_ HINSTANCE hPrevInstance,
+                     _In_ LPWSTR    lpCmdLine,
+                     _In_ int       nCmdShow)
+{
+    UNREFERENCED_PARAMETER(hPrevInstance);
+    UNREFERENCED_PARAMETER(lpCmdLine);
+
+    try {
+        rcbe::core::WindowManager wm(hInstance, L"MainWindow", nCmdShow);
+        rcbe::core::window_config wc{};
+        wc.caption = L"RCBE Engine";
+        wc.background_color = rcbe::visual::RGBAColor(0.5, 0.5, 0.5, 1.0);
+        wc.position = decltype(wc.position)(10.0, 10.0);
+        wc.size = rcbe::core::IntegralDimensions(1024, 768);
+        wc.process_input = true;
+        auto main_window = wm.createWindow(std::move(wc));
+
+        rcbe::core::EditorInputManager::HandlerCollection handlers = {
+            {
+                rcbe::core::InputEventType::left_button_press, 
+                [main_window](
+                    rcbe::core::InputManagerImplementation &im,
+                    rcbe::core::InputManagerImplementation::InputEventReference event,
+                    rcbe::core::InputManagerImplementation::PreviousEventReference previous
+                ) {
+                    auto x = GET_X_LPARAM(event.lparam);
+                    auto y = GET_Y_LPARAM(event.lparam);
+                    HDC hdc = main_window->getGraphicContext();
+                    COLORREF color = RGB(255,255,0);
+                    
+                    SetPixel(hdc, x, y, color); // SetPixel(HDC hdc, int x, int y, COLORREF color)
+             }},
+            {
+                rcbe::core::InputEventType::key_press,
+                [main_window](
+                    rcbe::core::InputManagerImplementation &im, 
+                    rcbe::core::InputManagerImplementation::InputEventReference event, 
+                    rcbe::core::InputManagerImplementation::PreviousEventReference previous
+                ) {
+                 if (main_window->getInputManager()->getValue(rcbe::core::KeyboardEventType::enter)) {
+                     rcbe::visual::texture_config tconf {};
+
+                     tconf.component_order = decltype(tconf.component_order)::GBR;
```

Comment:
Are you sure, component order is right?

Reply:
No, TGA parsing and work with image needs some major refactoring.